docs/developer-docs/setup/pulling-canister-dependencies.md

```diff
@@ -0,0 +1,430 @@
+
+#### Non-`pullable` examples:
+
+If the canister is meant for personal use and not intended for others to use, it does not make sense for the canister to be `pullable`. 
+
+If a canister wasm is published for other developers to use for deploying their own instance, it does not make sense for the canister to be `pullable` since the canister ID of the instance is not static; users can test integrations locally and deploy them directly. An example of this canister type is the asset canister generated by dfx. 
+
+## `Pullable` dfx.json example
+
+For a canister to be `pullable`, the `dfx.json` file must include a `pullable` definition. Below is an example:
+
+```json
+{
+  "canisters": {
+    "service": {
+      "type": "motoko",
+      "main": "src/pullable/main.mo",
+      "pullable": {
+        "dependencies": [],
+        "wasm_url": "https://github.com/lwshang/pullable/releases/latest/download/service.wasm",
+        "init_guide": "A natural number, e.g. 1"
+      }
+    }
+  }
+}
+```
+
+:::caution
+The wasm module of a `pullable` canister must be hosted via an URL so that service consumers can download it when pulling the dependency.
+
+GitHub Releases are a good, free option if the project is open source on GitHub. The GitHub URL schema is:
+
+`https://github.com/<USERNAME>/<REPONAME>/releases/latest/download/<FILENAME>`
+:::
+
+## Service provider workflow overview
+First, a service provider must configure a canister to be `pullable` by setting it as such in the `dfx.json` file. 
+
+An example of a provider `dfx.json` which has a `pullable` "service" canister can be found below:
+
+```json
+{
+    "canisters": {
+        "service": {
+            "type": "motoko",
+            "main": "src/main.mo",
+            "pullable": {
+                "wasm_url": "http://example.com/a.wasm",
+                "wasm_hash": "d180f1e232bafcee7d4879d8a2260ee7bcf9a20c241468d0e9cf4aa15ef8f312",
+                "dependencies": [
+                    "yofga-2qaaa-aaaaa-aabsq-cai"
+                ],
+                "init_guide": "A natural number, e.g. 10."
+            }
+        }
+    }
+}
+```
+
+The `pullable` object will be serialized as a part of the `dfx` metadata and attached to the wasm.
+
+To better understand the `pullable` object, let's look at each property in depth. 
+
+- `wasm_url`: A URL used to download the canister wasm module which will be deployed locally.
+- `wasm_hash` A SHA256 hash of the wasm module located at `wasm_url`. This field is optional. In most cases, the wasm module at `wasm_url` will be the same as the on-chain wasm module. This means that dfx can read the state tree to obtain and verify the module hash. In other cases, the wasm module at `wasm_url` is not the same as the on-chain wasm module. For example, the Internet Identity canister provides a `development` flavor to be integrated locally. In these cases, `wasm_hash` provides the expected hash, and `dfx` verifies the downloaded wasm against this.
```

Comment:
```suggestion
- `wasm_hash` A SHA256 hash of the wasm module located at `wasm_url`. This field is optional. In most cases, the wasm module at `wasm_url` will be the same as the on-chain wasm module. This means that dfx can read the state tree to obtain and verify the module hash. In some cases, the wasm module at `wasm_url` is not the same as the on-chain wasm module. For example, the Internet Identity canister provides a `development` variant to be integrated locally. In these cases, `wasm_hash` provides the expected hash, and `dfx` verifies the downloaded wasm against this.
```

I also think we should expand on the implication of not receiving the expected hash. Will dfx print a warning? Should developers be concerned? When should they ignore the warning, and in what circumstances should they heed the warning?

Reply:
Updated this bullet point with information from @lwshang via Slack DM conversation.